liquibase-core/src/main/java/liquibase/sqlgenerator/core/SetTableRemarksGenerator.java

```diff
@@ -37,44 +37,35 @@ public Sql[] generateSql(SetTableRemarksStatement statement, Database database,
 		} else if (database instanceof MSSQLDatabase) {
 			String schemaName = statement.getSchemaName();
 			if (schemaName == null) {
-				schemaName = database.getDefaultSchemaName();
-			}
-			if (schemaName == null) {
-				schemaName = "dbo";
+				schemaName = database.getDefaultSchemaName() != null ? database.getDefaultSchemaName() : "dbo";
 			}
+			String tableName = statement.getTableName();
+			String qualifiedTableName = String.format("%s.%s", schemaName, statement.getTableName());
 
-			sql = "DECLARE @TableName SYSNAME " +
-					"set @TableName = N'" + statement.getTableName() + "'; " +
-					"DECLARE @FullTableName SYSNAME; " +
-					"SET @FullTableName = N'" + schemaName+"."+statement.getTableName() + "';" +
-					"DECLARE @MS_DescriptionValue NVARCHAR(3749); " +
-					"SET @MS_DescriptionValue = N'" + remarksEscaped + "';" +
-					"DECLARE @MS_Description NVARCHAR(3749) " +
-					"set @MS_Description = NULL; " +
-					"SET @MS_Description = (SELECT CAST(Value AS NVARCHAR(3749)) AS [MS_Description] " +
-					"FROM sys.extended_properties AS ep " +
-					"WHERE ep.major_id = OBJECT_ID(@FullTableName) " +
-					"AND ep.name = N'MS_Description' AND ep.minor_id=0); " +
-					"IF @MS_Description IS NULL " +
-					"BEGIN " +
-					"EXEC sys.sp_addextendedproperty " +
-					"@name  = N'MS_Description', " +
-					"@value = @MS_DescriptionValue, " +
-					"@level0type = N'SCHEMA', " +
-					"@level0name = N'" + schemaName + "', " +
-					"@level1type = N'TABLE', " +
-					"@level1name = @TableName; " +
-					"END " +
-					"ELSE " +
-					"BEGIN " +
-					"EXEC sys.sp_updateextendedproperty " +
-					"@name  = N'MS_Description', " +
-					"@value = @MS_DescriptionValue, " +
-					"@level0type = N'SCHEMA', " +
-					"@level0name = N'" + schemaName + "', " +
-					"@level1type = N'TABLE', " +
-					"@level1name = @TableName; " +
-					"END";
+			sql = "IF EXISTS( " +
+					" SELECT extended_properties.value" +
+					" FROM SYS.EXTENDED_PROPERTIES" +
+					" WHERE major_id = OBJECT_ID('" + qualifiedTableName + "')" +
+					" AND name = N'MS_DESCRIPTION'" +
+					" AND minor_id = 0" +
+					" )" +
+					" BEGIN " +
+					" EXEC sys.sp_updateextendedproperty @name = N'MS_Description'" +
+					" , @value = N'" + remarksEscaped + "'" +
+					" , @level0type = N'SCHEMA'" +
+					" , @level0name = N'" + schemaName + "'" +
+					" , @level1type = N'TABLE'" +
+					" , @level1name = N'" + tableName + "'" +
+					" END " +
+					" ELSE " +
+					" BEGIN " +
+					" EXEC sys.sp_addextendedproperty @name = N'MS_Description'" +
+					" , @value = N'" + remarksEscaped + "'" +
+					" , @level0type = N'SCHEMA'" +
+					" , @level0name = N'" + schemaName + "'" +
+					" , @level1type = N'TABLE'" +
+					" , @level1name = N'" + tableName + "'" +
+					" END";
```

Comment:
Can you summarize the changes you made to this block of SQL?

Reply:
Sure! I removed the query variables from the query. Originally would set up a table name variable `DECLARE @TableName SYSNAME; SET @TableName = N'BLAH';` I have removed this in favor of directly using the table name etc in the query. 
So before we had: 
```
DECLARE @TableName SYSNAME
set @TableName = N'person';
DECLARE @FullTableName SYSNAME;
SET @FullTableName = N'dbo.person';
DECLARE @MS_DescriptionValue NVARCHAR(3749);
SET @MS_DescriptionValue = N'A String';
DECLARE @MS_Description NVARCHAR(3749)
set @MS_Description = NULL;
SET @MS_Description = (SELECT CAST(Value AS NVARCHAR(3749)) AS [MS_Description]
                       FROM sys.extended_properties AS ep
                       WHERE ep.major_id = OBJECT_ID(@FullTableName)
                         AND ep.name = N'MS_Description'
                         AND ep.minor_id = 0);
IF @MS_Description IS NULL
    BEGIN
        EXEC sys.sp_addextendedproperty @name = N'MS_Description', @value = @MS_DescriptionValue,
             @level0type = N'SCHEMA', @level0name = N'dbo', @level1type = N'TABLE', @level1name = @TableName;
    END
ELSE
    BEGIN
        EXEC sys.sp_updateextendedproperty @name = N'MS_Description', @value = @MS_DescriptionValue,
             @level0type = N'SCHEMA', @level0name = N'dbo', @level1type = N'TABLE', @level1name = @TableName;
    END;
```
Now it looks like: 
```
IF EXISTS(SELECT extended_properties.value
          FROM SYS.EXTENDED_PROPERTIES
          WHERE major_id = OBJECT_ID('dbo.person')
            AND name = N'MS_DESCRIPTION'
            AND minor_id = 0)
    BEGIN
        EXEC sys.sp_updateextendedproperty @name = N'MS_Description', @value = N'A String', @level0type = N'SCHEMA',
             @level0name = N'dbo', @level1type = N'TABLE', @level1name = N'person'
    END
ELSE
    BEGIN
        EXEC sys.sp_addextendedproperty @name = N'MS_Description', @value = N'A String', @level0type = N'SCHEMA',
             @level0name = N'dbo', @level1type = N'TABLE', @level1name = N'person'
    END;
```
The reason I had to change this is we generate table comments and table column comments together. The repeated usage of the same `@TableName` even when redeclared throws errors.